packages/tunnel/src/commands/deploy/utils.ts

```diff
@@ -0,0 +1,154 @@
+import AdmZip from 'adm-zip';
+import chalk from 'chalk';
+
+import { consoleLog } from '../../utils.js';
+
+type TokenResponse = {
+  access_token: string;
+  token_type: string;
+  expires_in: number;
+  scope: string;
+};
+
+type DeployArgs = {
+  appId: string;
+  appSecret: string;
+  endpoint: string;
+  experiencePath: string;
+  verbose: boolean;
+};
+
+export const deployToLogtoCloud = async ({
+  appId,
+  appSecret,
+  endpoint,
+  experiencePath,
+  verbose,
+}: DeployArgs) => {
+  if (verbose) {
+    consoleLog.info('Zipping files...');
+  }
+  const zipBuffer = await zipFiles(experiencePath);
```

Comment:
shall we support zip files as well?

Reply:
Hmmm, I think maybe not. Can't think of a scenario.

Reply:
actually there's one scenario: you don't want to include some files e.g. `.git` when zipping. having a zip option can provide the flexibility to the developer.

our first step can be filtering out some built-in file patterns though.

Reply:
Will improve this in a follow-up PR